File: common/src/main/java/com/wdz/common/mvvm/BaseMvvmViewModel.java
package com.wdz.common.mvvm;

import android.content.Context;

import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.wdz.common.net.HttpRequestStatus;

public abstract class BaseMvvmViewModel<M extends BaseModel> extends ViewModel{

    protected M model;
    protected LifecycleOwner lifecycleOwner;
    /**
     * http请求时的状态变化liveData
     */
    protected MutableLiveData<HttpRequestStatus> httpLiveData = new MutableLiveData<>();

    protected abstract void initModel(Context context);

    /**
     * 设置lifecycleOwner
     * @param lifecycleOwner
     */
    protected void setLifecycleOwner(LifecycleOwner lifecycleOwner){
        this.lifecycleOwner = lifecycleOwner;
    };

    protected LifecycleOwner getLifecycleOwner(){
        return lifecycleOwner;
    }

    public MutableLiveData<HttpRequestStatus> getHttpLiveData(){
        return httpLiveData;
    }

}

File: kt_common/src/main/java/com/wdz/ktcommon/base/BaseMvvmViewModel.kt
package com.wdz.ktcommon.base

import android.app.Application
import android.content.Context
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.wdz.ktcommon.http.HttpRequestStatus
import com.wdz.ktcommon.http.repository.NetRepository

/**

 * @Author dezhi.wang

 * @Date 2021/3/31 13:38

 */
abstract class BaseMvvmViewModel<M: BaseModel>(application: Application) : AndroidViewModel(application) {

    protected var model: M? = null


    /**
     * http请求时的状态变化liveData
     */
    var httpLiveData: MutableLiveData<HttpRequestStatus> =
        MutableLiveData()

    protected abstract fun initModel(context: Context)


}